### ISSUE TITLE
Comparison with paste.js / join forces

### ISSUE THREAD

**Issue by dandv** [vladmalik/pasteimage#5]
How does this plugin compare with @layerssss [paste.js](https://github.com/layerssss/paste.js)? Since they're both jQuery plugins, maybe join forces?

**Comment by vladmalik** [comment]
Looked it over. Similar method. Major difference is I only care about images and I output a data URI. The URI can be uploaded directly to the server, shared across browser sessions, and so on just like any image URL. The other plugin outputs a Blob object, which persists for the duration of the session. Not sure how it works with uploading blobs. Also, dataURI support goes back to IE8. Blob support is pretty good too though. Not sure about advantages and disadvantages.

**Comment by vladmalik** [comment]
Actually, looks like it's pretty much identical. It just looks like he has more checks in place. Might retire mine in favor of his. Going to check into advantages of his.

**Comment by dandv** [comment]
Awesome! I'll keep an eye on your findings. I'm interested in [adding image pasting support to Trumbowyg](https://github.com/Alex-D/Trumbowyg/issues/135).

**Comment by layerssss** [comment]
hi @dandv , vladmalik is right, we used pretty the same tricks to get the image from clipboard, in paste.js I also added IE's hacks, and there are three invoking methods relatively for `div`, `textarea`, `[contenteditable]`. You really should check out that third one! :) . I've also listed the details of the three modes in https://github.com/layerssss/paste.js/issues/14

Neither of the two plugins can currently handle images pasted into Safari, it's been a pain for me for a long time.  But I don't think this can be solved on current version of Safari, unless [this issue](https://bugs.webkit.org/show_bug.cgi?id=49141) been dealt with.

**Comment by vladmalik** [comment]
Also see this comment: https://github.com/layerssss/paste.js/issues/14#issuecomment-119052389

**Comment by dandv** [comment]
@layerssss: I don't have a Safari to test, but apparently [Google managed to paste images](https://productforums.google.com/forum/#!topic/gmail/LmxLVYb0Ofk)? How does GitHub do it (or they don't support pasting images from Safari?)

**Comment by layerssss** [comment]
@dandv no, google does not. An `<img>` tag with `src` set to `webkit-fake-url:...` was inserted into the email, but will not be readable by anyone.

**Comment by vladmalik** [comment]
@layerssss There might be a way to do it with flash (similarly to how many sites copy selected text into your clipboard)?

**Comment by layerssss** [comment]
@vladmalik , yeah, there should be, but I havn't  worked with flash before, might worth a while to look into it.
